File: server/Models/Employee.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using server.Models;
namespace server.Models
{
    public class Employee
    {
        [Key]
        public long Id { get; set; }

        // FK
        public long UserId { get; set; }
        public User User { get; set; } = null!;

        public long RoleId { get; set; }
        public EmployeeRole Role { get; set; } = null!;

        public DateTime StartDate { get; set; }
        public DateTime? EndDate { get; set; }

        [Column(TypeName = "decimal(15,2)")]
        public decimal Salary { get; set; }

        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
    }
}

File: kkn_backend/app/models/employee.go
package models

import (
	"github.com/goravel/framework/database/orm"
)

type Employee struct {
	orm.Model

	UserId uint64
	User   User

	RoleId uint64
	Role   EmployeeRole

	StartDate string
	EndDate   *string
	Salary    float64
}
